### DISCUSSION TITLE
WordPress — Vagas de Emprego em Queda?

### DISCUSSION THREAD

**Discussion by renanlara** [frontendbr/forum#904] [Discussão]
Boa tarde galerinha, tudo bem?
Pessoal, faz alguma tempo que eu acompanho o [FrontVagas](https://github.com/frontendbr/vagas) e o site do [Trampos](https://trampos.co/) e percebi que vagas para desenvolvimento em WordPress diminuiu muito.

A demanda de profissionais que manjam muito de JS realmente aumentaram muito, mas o WordPress foi um queridinho das agências.

O que de fato tem acontecido? O que vocês pesam sobre o assunto?

**Comment by luizbills** [comment]
IMHO A demanda ainda é muito alta, pois wordpress é uma solução muito barata (em termos de contratação). Porém estamos em momento de transição de tecnologias.

**Comment by viniciusdocanto** [comment]
Tenho percebido isso já tem um ano, que vagas com wordpress tem sumido do mercado, entrando muita vaga com Angular/React

**Comment by AmandaFonseca** [comment]
Eu só trabalhei 1 vez com wordpress na criação de um blog para uma empresa, depois que eu já tinha entregue esse trabalho descobri que agora o wordpress tem 1 API REST ou seja qualquer desenvolvedor que manje de alguma linguagem de programação pode fazer um CRUD usando o GET ou POST sem ter conhecimento de PHP. Não sei se isso realmente tem a ver com a queda de profissionais que trabalham com Wordpress (até porque estou começando a trabalhar/estudar agora  e eu não tenho uma looonga experiência e visão de mercado e tal) mas  a API REST possibilita que outros devs trabalhem com Wordpress.

**Comment by fernahh** [comment]
Acredito que os frameworks back-end, como o Rails, chegaram na ápice da praticidade e produtividade. Além disso, o Wordpress sempre foi bastante usado para criação de sites institucionais, blogs, e que hoje são bem resolvidos com geradores estáticos.

Também existem várias ferramentas de "CMS as a service", como o Contentful. Até mesmo vários providers serverless.

Em resumo: hoje existem várias soluções, que juntas, entregam o mesmo (ou até mais) valor que sites wordpress, e por consequência diminuiu a procura por profissionais no mercado.

**Comment by barbier** [comment]
De acordo com pesquisas™ o [Wordpress é usado em quase 30% da internet](https://w3techs.com/). Talvez o cenário brasileiro de alguma forma tenha estagnado ou as fábricas/agências se estabilizaram na forma de trabalhar, mas eu não diria que o Wordpress está morrendo....

**Comment by bernardodiasc** [comment]
Alguns anos atrás o WP era usado em 60%~70%, parece que ta caindo bem. (Mas não tenho fontes desses números, lembro de ter lido algumas pesquisas na época, mas nem da pra confiar direito mesmo)

**Comment by guilhermealveslopes** [comment]
Wordpress é muito prático, principalmente por ser como já disseram ai, o queridinho das agências.
Muito usuário final ao invés de gastar com desenvolvimento, compra um tema e coloca o site no ar.
Com isso e o gigantesco crescimento de empreendedores precisando de um projetinho na web, deram muita popularidade pro WP.

Mas quando a aplicação é maior, o WP já deixa a desejar em vários pontos, ai talvez a queda atual. 
Pois certamente, sempre aparece algo mais inovador.

**Comment by mksanti** [comment]
Acredito que não esteja procurando no lugar certo. Se acompanhar a comunidade WP vai ver que rola vagas sempre. Sem contar os sites de emprego específicos para WP como [hangar](https://hangarwp.com/) ou o proprio [WP Jobs](https://jobs.wordpress.net/).
Por ai comentaram sobre a WP API Rest e acredito que este modulo tenha aumentado a demanda e não o contrario. Já que agora temos todas as funcionalidade e a dos seus plugins (woocommerce, ACF, membership, Buddypress) disponíveis via API o que facilita a integração com qualquer outra linguagem, sem contar que se você quer personalizar ainda mais o WP seja via API ou não tem que sacar de PHP e do [Codex](https://codex.wordpress.org/).

**Comment by LamegoMKTDigital** [comment]
Pessoal boa tarde.

Para quem tiver interesse em vaga em WordPress e quiser fazer uma renda extra, segue link do nosso programa de contratações http://mktdigital.com.vc/contratacao/

Atenciosamente, 
Equipe Lamego MKT Digital.

**Comment by renanlara** [comment]
Caracas @LamegoMKTDigital, vocês são demais! Me inscrevi na vaga de vocês, fui aprovado e agora vocês comentam em uma issue que eu criei. Que sensacional!

**Comment by felipefialho** [comment]
Que foda!

**Comment by allysonsouza** [comment]
O WordPress nunca chegou à 60% da web não @bernardodiasc , tá batendo 30% esse ano, e vem crescendo. É a plataforma líder na web já há alguns anos e continua crescendo, coisa difícil em qualquer mercado, se manter crescendo enquanto já está na liderança.

O WooCommerce é a maior plataforma de desenvolvimento de lojas virtuais, se não me engano representa 40% da lojas.

Mas e então, o que acontece com o Brasil? Eu gostaria de saber também. Aqui parece que desenvolvemos uma grande aversão pelo WordPress para grandes projetos e desenvolvemos uma cultura de apenas personalização de temas e instaladores de plugins, subutilizando a plataforma. No campo do e-commerce a desconfiança por software livre é grande e a VTEX vem abocanhando o mercado com sua expertise e sistema de afiliação de agências. 

Falta interesse e mente aberta para a plataforma, estamos em 2018 e na comunidade WordPress ainda temos que ler o texto pronto para quando pensam que o WordPress só é para blog, ou não é seguro, etc. E dos desenvolvedores falta especialização, a cultura de instaladores de temas e plugins das agências fez com que não se criasse desenvolvedores que trabalham com WordPress. Mas este ano será lançado o Gutenberg, novo editor de conteúdo, e muitas outras novidades, amadurecimento do WooCommerce com a versão 3.0 na direção de suportar melhor lojas grandes, então potencial tem, vamos ver como o Brasil segue nisso... Fora os eventos, comunidade está ativa e já tem WordCamp Florianópolis marcado esse ano!

**Comment by luizbills** [comment]
vale acrescentar também ao comentário do @allysonsouza que a Google está juntando uma equipe para ajudar no desenvolvimento do core do wordpress, afim melhorar a performance do wp.

**Comment by lincolnlemos** [comment]
IMHO, Trabalho com WP há 7 anos e minha percepção é que as ofertas só aumentam. Meu foco atualmente é atender agencias digitais que desejam terceirizar o desenvolvimento dos sites em WordPress com qualidade e segurança, sendo assim estou meio por fora de vagas para CLT, mas pra freela vejo muita coisa.

A oferta é tão grande que vejo muito freelancer entregando porcaria e queimando prazo porque não esta preocupado em ficar sem job. Semana passada estava conversando com um parceiro que trabalha atualmente na Espanha e ele disse "...é tanta oferta que nego nem liga mais.... Caga no pau mesmo que tem mais um monte querendo...". No momento estávamos conversando sobre um projeto no qual o freelancer "sumiu" no meio do projeto e agora precisa entrar alguém pra fazer o salva vidas.

Como já dito anteriormente e muito bem citado no podcast do hipsters.tech "#72 Por trás do WordPress", existe uma facilidade muito grande de entrada para iniciar no WordPress. Desde a curva de aprendizado, tecnologias envolvidas e até mesmo a infra necessária. Nesse sentido, concordo com o @allysonsouza  em dizer que tem muito personalizadores de temas e instaladores de plugins. Ao sentido contrário dessa turma, existem soluções como o: vip.wordpress.com, uma solução empresarial robusta para grandes empresas; e milhares de devs aperfeiçoando plugins, temas e até mesmo o core desse software fabuloso :D 

IMHO, essa relação de porcentagem do total de sites que usam WordPress não serve como parâmetro. A web vem evoluindo a cada dia e as razões por que se faz um "site" estão cada vez mais distintas. O WordPress sempre atendeu muito bem seu público, e pelo o que sua trajetória mostrou, ele sempre aumentou sua gama de features, consequentemente aumentando seu público alvo.

Você tem a fonte disso @luizbills ?

**Comment by luizbills** [comment]
@lincolnlemos
https://medinathoughts.com/2018/01/29/wordpress-google/
https://torquemag.io/2017/12/google-and-wordpress/

**Comment by helioricardo** [comment]
O WP não para de crescer... Com o 5 teremos uma nova revolução, pois o WP se encontra com o React e vão gerar o Gutenberg... Por falar nisso, se gosta de WP e JS avançado a BS é a sua casa.

## Descrição da vaga

Vaga para dev pleno WP com foco em front e JS avançado

## Local

Ficamos na WeWork Carioca, Rio de Janeiro - Centro

## Benefícios

- VR
- VT
- Outros

#### Diferenciais

Somos flexíveis quanto à horário de trabalho, vestimenta (bermudas permitidas)... E isso é só o começo da conversa...

Ótimo ambiente no escritório. Já falamos do choppinho zero-800, cafézinho, diversos chás, água saborizada, ping pong, totó, eventos semanais e da vista incrível?

## Requisitos

**Pessoal:**
- Auto-gerenciamento e proatividade para lidar com demandas e prioridades num ambiente ágil
- Ame código limpo e segue boas práticas de programação
- Estime o tempo do seu próprio trabalho e levante riscos quando solicitado
- Trabalhe de forma integrada com o time de design
- Curioso por natureza e que ame desafios
- Tenha foco no usuário final

**Obrigatórios:**
- PHP/WP (Conhece Pods, Carbon Fields, Rest API do WP? Estamos alinhados)
- Tradução PSDs para HTML sem nenhum problema
- JS avançado. ES6...
- Git
- SASS
- Alto conhecimento em HTML5 e CSS3 - Dããã
- Confortável com linha de comando
- Consumo de APIs RESTful
- Conhecimento de mecanismos de autenticação, como JWT
- BaaS (Parse, Firebase, Graphql Cool...)

**Desejáveis:**
- React/Redux
- Gulp

**Diferenciais (para impressionar):**
- React Native
- Laravel
- Projetar e construir aplicações distribuídas usando arquitetura de micro-serviços
- ElasticSearch
- MongoDB
- Figma
- WebPack
- ES8

## Contratação

PF ou PJ a combinar

## Nossa empresa

[BSCONSULT - web & mobile dev.](http://www.bsconsult.com.br)

A BS planeja e desenvolve websites, webapps e apps para agências, startups, pequenas e médias empresas, sempre focando no cliente final e na qualidade técnica do projeto.


## Como se candidatar

Por favor envie um email para vagas@bsconsult.com.br com seu CV anexado - enviar no assunto: Vaga WP+JS.
